Client/Frontend/Browser/TabTrayController.swift

```diff
@@ -630,16 +630,16 @@ class TabTrayController: UIViewController {
     }
 
     private func openNewTab(request: NSURLRequest? = nil) {
-        toolbar.addTabButton.userInteractionEnabled = false
+        toolbar.userInteractionEnabled = false
 
         // We're only doing one update here, but using a batch update lets us delay selecting the tab
         // until after its insert animation finishes.
         self.collectionView.performBatchUpdates({ _ in
             let tab = self.tabManager.addTab(request, isPrivate: self.privateMode)
             self.tabManager.selectTab(tab)
         }, completion: { finished in
+            self.toolbar.userInteractionEnabled = true
             if finished {
```

Comment:
This looks pretty suspicious to me -- I'd bet that this is the problem.

The `completion` callback is called when the updates are done, and the `finished` flag given to the callback tells us whether the animation finished. We don't actually care whether the animation finished, though; we only care that the updates happened, which means we want to hide the view controller. In other words, even if the animation was interrupted and `finished` is `false` here (which is probably what happens when menu is tapped), we still want to pop the view controller when we're done; otherwise, it'll never get removed.

I see we do similar things in other places of our code (see `SELdidTogglePrivateMode` in this same file). I'm pretty sure we should remove the `if finished` condition in all of these -- the post-animation completion code shouldn't be dependent on whether the animation was or wasn't interrupted.